src/lwip.c

```diff
@@ -119,6 +136,11 @@ void MX_LWIP_Init(void) {
 
     /* USER CODE BEGIN 3 */
     netif_set_link_callback(&eth0, netif_link_callback);
+    netif_set_status_callback(&eth0, netif_status_callback);
```

Comment:
netif_set_status_callback documentation says "Set callback to be called when interface is brought up/down or address is changed while up". may be  netif_set_link_callback is not needed when you use this?
otherwise both callback may be called, just check

Reply:
Both callbacks are needed. There need to be a callback that will automaticaly set interface up, when I plug the cabel in. I did not find any situations when they both are called. Even if there are some, when interface is already set down, netif_set_down(&eth0) function has no efect.